### PR TITLE
Demo chat

### PR DESCRIPTION
**Pull request by LevilkTheReal**
Core changes:
- Instead of storing each user on a separate index on the graffiti feed, (to reduce read latency). All of the users are stored in an array on each index.
- I wired in MetaMask. Now the users who wants to join has to register through a modal with the necessary chat details (name, key, stamp). One account can be registered once - if the connected account is registered then it's a rejoin.
- Event based state changes
- I introduced two async queue. One for the users, on for the messages to to prevent the overload of the node.
- Initalization phase for users to check if the current account is already registered and to speed up the first messages
- Only start showing messages from the latest indexex
- latest bee, bee-js updgrade
- close to latest stream changes merge

Refactors:
 - I removed local store. We want to reduce the usage of any kind of browser store. It's hard to maintain, hard to follow, easy to create async issues. I found a cleaner solution.
 - I removed the context. Same reason as above.
 - Renamed ambiguous vars
 - Minor changes to follow current conventions
 - Fixed and optimized react comps which caused  a couple of render loops
 - Fixed couple of DIYs and did some additional modularizations
 
 Knowns issues:
 - We don't want to use MUI. It's hard and not clean to customize, also heavily increases the bundle size. Our own comp lib is on the way, we gonna switch to that soon
 - Buggy, and lack of error handling
 - Current solution is not ready to be a separate lib
 - Key management handling is a must, we can't store keys anywhere
 - Types should be refactored
 
 - The current solution is scaling really badly. The main limitation is the for loop that has to check the new messages.
 Let's assume that we have 8 users. Then we have to check the 8 user's feed every time in a loop. We only see the last message of the 8th user at the end of the loop. The fact that 404s are slow requests make the situation worse. If only the 8th user has a message the rest not then we have to wait 7*~5sec to reach the 8th user in best case scenario.
 
We have to make the 404 (no update) a fast call. Close to 0 as possible but minimum as fast a chunk request.  /  Rethink the current solution and remove the message loop and store everything on a single feed in a secure way.

It might worth exploring the use of service workers to utalize multiple threads. We are doing so much requests that the browser's event queue easily can get stuck/blocked.

### CODE REVIEW

File: src/pages/Search/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { ControllerTextInput } from '../../components/TextInput/ControllerTextInput';
+import { WithErrorBoundary } from '../../hooks/WithErrorBoundary';
+import { assertAtLeastFourChars, assertPublicAddress } from '../../utils/formValidation';
+
+import './Search.scss';
+
+interface FormData {
+  walletAddress: string;
+  streamTopic: string;
+  /*   timeslice: string;
```
Comment (aranyia) [inline src/pages/Search/Search.tsx:15]: These are here for reference or will still be needed?

File: src/pages/Search/Search.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    defaultValue: '',
+    rules: { required: 'Wallet address is required', validate: assertPublicAddress },
+  },
+  {
+    name: 'streamTopic',
+    label: 'This is how others will find your stream',
+    defaultValue: '',
+    rules: { required: 'Topic is required', validate: assertAtLeastFourChars },
+  },
+  /*   {
```
Comment (aranyia) [inline src/pages/Search/Search.tsx:35]: Will these be relevant in the future as part of the code?

File: src/pages/Stream/Stream.tsx [hunk 3]
```diff
@@ -1,195 +1,137 @@
...
-          : undefined,
-      },
-    );
 
-    // We save chatWriter to state, Graffiti-feed connector will be re-generated every time it is used (nothing needs to be saved)
-    const result = await initChatRoom(feedDataForm.topic.value, feedDataForm.stamp.value as BatchId);
-    if (!result) throw 'initChatRoom gave back null';
-    
+    /*     await startStream({ address: account!, key: data.key }, data.streamTopic, data.stamp as BatchId, {
+      audio: options.audio,
```
Comment (aranyia) [inline src/pages/Stream/Stream.tsx:78]: These will be part of the code or documentation?

File: src/utils/webm.ts [hunk 4]
```diff
@@ -24,21 +26,32 @@ export function parseVint(buffer: Uint8Array, offset = 0) {
   }
 
   return {
-    length: lengthOfLengthDescriptor,
+    lengthOfLengthDescriptor,
```
Comment (aranyia) [inline src/utils/webm.ts:29]: I think I get what this variable is for, but a clearer name would be better if can come up with a good one.

File: src/pages/Search/Search.tsx (hunk 1 above)
Comment (LevilkTheReal) [reply src/pages/Search/Search.tsx:15]: I removed these but we will need them later ( I left them here as a reference)

File: src/pages/Search/Search.tsx (hunk 2 above)
Comment (LevilkTheReal) [reply src/pages/Search/Search.tsx:35]: I removed these but we will need them later ( I left them here as a reference)

File: src/pages/Stream/Stream.tsx (hunk 3 above)
Comment (LevilkTheReal) [reply src/pages/Stream/Stream.tsx:78]: These for commented out for testing purposes
